### ISSUE TITLE
Setting `moduleSideEffects: false` removes side effectful getter inside a exported function since 4.9.2

### ISSUE THREAD

**Issue by sapphi-red** [rollup/rollup#5408]
### Rollup Version

4.12.0 (since 4.9.2)

### Operating System (or Browser)

StackBlitz

### Link To Reproduction

https://stackblitz.com/edit/rollup-repro-3brsqu?file=src%2FsupportsPassive.js,dist%2Fmain.js

### Expected Behaviour

`supportsPassive` is not removed.
```js
const supportsPassive = (() => {
  // simular to jQuery's test
  let supported = false;
  try {
    addEventListener(
      'test',
      () => {
        // do nothing
      },
      Object.defineProperty({}, 'passive', {
        get() {
          supported = true;
          return true;
        },
      })
    );
  } catch (e) {
    // do nothing
  }
  return supported;
})();

class TouchBackendImpl {
  addEventListener() {
    const options = supportsPassive ? { capture, passive: false } : capture;
    console.log(options);
  }
}

export { TouchBackendImpl };
```

### Actual Behaviour

`supportsPassive` is removed;
```js
class TouchBackendImpl {
  addEventListener() {
    const options = capture;
    console.log(options);
  }
}

export { TouchBackendImpl };
```

### Additional Information

This happens with 4.9.2 and doesn't happen with 4.9.1. I guess #5322 is related.

Original issue: https://github.com/vitejs/vite/issues/16065

**Comment by TrickyPi** [comment]
The more concise reproducible case: https://rollupjs.org/repl/?version=4.12.0&shareable=JTdCJTIyZXhhbXBsZSUyMiUzQW51bGwlMkMlMjJtb2R1bGVzJTIyJTNBJTVCJTdCJTIyY29kZSUyMiUzQSUyMmltcG9ydCUyMCU3QiUyMHN1cHBvcnRzUGFzc2l2ZSUyMCU3RCUyMGZyb20lMjAnLiUyRm1hdGhzLmpzJyUzQiU1Q24lNUNuZXhwb3J0JTIwY2xhc3MlMjBUb3VjaEJhY2tlbmRJbXBsJTIwJTdCJTVDbiUyMCUyMGNvbnN0cnVjdG9yKCklMjAlN0IlNUNuJTIwJTIwJTIwJTIwY29uc3QlMjBvcHRpb25zJTIwJTNEJTIwc3VwcG9ydHNQYXNzaXZlKCklMjAlM0YlMjAlN0IlMjBjYXB0dXJlJTJDJTIwcGFzc2l2ZSUzQSUyMGZhbHNlJTIwJTdEJTIwJTNBJTIwY2FwdHVyZSUzQiU1Q24lMjAlMjAlMjAlMjBjb25zb2xlLmxvZyhvcHRpb25zKSUzQiU1Q24lMjAlMjAlN0QlNUNuJTdEJTVDbiUyMiUyQyUyMmlzRW50cnklMjIlM0F0cnVlJTJDJTIybmFtZSUyMiUzQSUyMm1haW4uanMlMjIlN0QlMkMlN0IlMjJjb2RlJTIyJTNBJTIybGV0JTIwc3VwcG9ydGVkJTIwJTNEJTIwZmFsc2UlM0IlNUNuJTVDbmJiYigoKSUyMCUzRCUzRSUyMCU3QiU1Q24lMjAlMjBzdXBwb3J0ZWQlMjAlM0QlMjAhc3VwcG9ydGVkJTNCJTVDbiU3RCklM0IlNUNuJTVDbmV4cG9ydCUyMGNvbnN0JTIwc3VwcG9ydHNQYXNzaXZlJTIwJTNEJTIwKCklMjAlM0QlM0UlMjAlN0IlNUNuJTIwJTIwcmV0dXJuJTIwc3VwcG9ydGVkJTNCJTVDbiU3RCU1Q24lMjIlMkMlMjJpc0VudHJ5JTIyJTNBZmFsc2UlMkMlMjJuYW1lJTIyJTNBJTIybWF0aHMuanMlMjIlN0QlNUQlMkMlMjJvcHRpb25zJTIyJTNBJTdCJTIyb3V0cHV0JTIyJTNBJTdCJTIyZm9ybWF0JTIyJTNBJTIyZXMlMjIlN0QlMkMlMjJ0cmVlc2hha2UlMjIlM0ElN0IlMjJtb2R1bGVTaWRlRWZmZWN0cyUyMiUzQWZhbHNlJTdEJTdEJTdE

**Comment by lukastaegert** [comment]
It seems that in general it does not track mutations when moduleSideEffects are `false`. Let's see what is behind this.

https://rollupjs.org/repl/?version=4.12.0&shareable=JTdCJTIyZXhhbXBsZSUyMiUzQW51bGwlMkMlMjJtb2R1bGVzJTIyJTNBJTVCJTdCJTIyY29kZSUyMiUzQSUyMmltcG9ydCUyMCU3QiUyMG1vZGlmaWVkJTIwJTdEJTIwZnJvbSUyMCcuJTJGb3RoZXIuanMnJTNCJTVDbiU1Q25jb25zdCUyMHJlc3VsdCUyMCUzRCUyMG1vZGlmaWVkJTIwJTNGJTIwJ09LJyUyMCUzQSUyMCdGQUlMJyUzQiU1Q25hc3NlcnQuc3RyaWN0RXF1YWwocmVzdWx0JTJDJTIwJ09LJyklM0IlMjIlMkMlMjJpc0VudHJ5JTIyJTNBdHJ1ZSUyQyUyMm5hbWUlMjIlM0ElMjJtYWluLmpzJTIyJTdEJTJDJTdCJTIyY29kZSUyMiUzQSUyMmV4cG9ydCUyMGxldCUyMG1vZGlmaWVkJTIwJTNEJTIwZmFsc2UlM0IlNUNubW9kaWZpZWQlMjAlM0QlMjB0cnVlJTNCJTIyJTJDJTIyaXNFbnRyeSUyMiUzQWZhbHNlJTJDJTIybmFtZSUyMiUzQSUyMm90aGVyLmpzJTIyJTdEJTVEJTJDJTIyb3B0aW9ucyUyMiUzQSU3QiUyMm91dHB1dCUyMiUzQSU3QiUyMmZvcm1hdCUyMiUzQSUyMmVzJTIyJTdEJTJDJTIydHJlZXNoYWtlJTIyJTNBJTdCJTIybW9kdWxlU2lkZUVmZmVjdHMlMjIlM0FmYWxzZSU3RCU3RCU3RA==

**Comment by lukastaegert** [comment]
Sorry, after starting work on this issue I dropped the ball and forgot about it. Here is the fix: #5658
